### PR TITLE
Remove the model package

### PR DESCRIPTION
**Pull request by frrist**
finally!

### CODE REVIEW

File: cmd/cli/docker/docker_run_cli_test.go
Comment (wdbaruni) [inline cmd/cli/docker/docker_run_cli_test.go:188]: why are we coupling `model` removal with supporting deprecated flags? Can we please do them separately, and only remove support of deprecated flags by 1.5.0?

File: pkg/node/config_defaults.go
Comment (wdbaruni) [inline pkg/node/config_defaults.go:46]: maybe you should've just left it and migrated it to `models.NoJobTimeout` instead of doing this in many places. I am also thinking we should change this to `-1` to indicate no timeout instead of max value

File: pkg/orchestrator/event_emitter.go
Comment (wdbaruni) [inline pkg/orchestrator/event_emitter.go:36]: can you please add a TODO to deprecate `models.JobEvent` and this event emitter? I don't think it is used anywhere except for logging, and it should be replaced by a jobstore watcher for history events

**Review by wdbaruni — CHANGES_REQUESTED** [review]
Nice work. Just a minor comment about decoupling removal of `model` pkg from removing support of legacy cli flags.

Also I recommend we merge this after 1.4.0 as it is a big change and bugs might slip

File: pkg/node/config_defaults.go
Comment (frrist) [reply pkg/node/config_defaults.go:46]: I think somewhere we have stated that `0` implies no timeout, wdyt of that pattern?

File: pkg/orchestrator/event_emitter.go
Comment (frrist) [reply pkg/orchestrator/event_emitter.go:36]: I'd love to remove these now, if we don't think they are being used for anything can I remove them now?

File: cmd/cli/docker/docker_run_cli_test.go
Comment (frrist) [reply cmd/cli/docker/docker_run_cli_test.go:188]: I can revert this change and land it later, tracking via https://github.com/bacalhau-project/bacalhau/issues/4182

File: pkg/node/config_defaults.go
Comment (wdbaruni) [reply pkg/node/config_defaults.go:46]: as it stands, 0 means no timeout was defined and will be replaced by default value. I don't know how 0 timeout and 0 default timeout is handled in our system. Also 0 queue-timeout means no queueing enabled.

I'll be concerned about trying to change the meaning of 0 and -1 timeout as part of this change, but just having a `NoJobTimeout` const or var seemed a good approach to easily change things in the future

File: pkg/orchestrator/event_emitter.go
Comment (wdbaruni) [reply pkg/orchestrator/event_emitter.go:36]: My preference is to do it on a separate PR. This will allow you to merge this PR today, which Udit is blocked on for the python sdk work, and will simplify the review of removing the `JobEvent` models

File: pkg/node/config_defaults.go
Comment (frrist) [reply pkg/node/config_defaults.go:46]: Okay sounds good, will define a default constant and use everywhere.

File: pkg/orchestrator/event_emitter.go
Comment (frrist) [reply pkg/orchestrator/event_emitter.go:36]: Sounds good.

**Review by wdbaruni — APPROVED** [review]

**Pull request merged**
